src/Umbraco.Web/GridTemplateExtensions.cs

```diff
@@ -26,18 +23,20 @@ public static MvcHtmlString GetGridHtml(this HtmlHelper html, IPublishedContent
 
         public static MvcHtmlString GetGridHtml(this HtmlHelper html, IPublishedContent contentItem, string propertyAlias)
         {
-            if (string.IsNullOrWhiteSpace(propertyAlias)) throw new ArgumentNullOrEmptyException(nameof(propertyAlias));
+            if (propertyAlias == null) throw new ArgumentNullException(nameof(propertyAlias));
+            if (string.IsNullOrWhiteSpace(propertyAlias)) throw new ArgumentException("Value can't be empty or consist only of white-space characters.", nameof(propertyAlias));
 
             return html.GetGridHtml(contentItem, propertyAlias, "bootstrap3");
         }
 
         public static MvcHtmlString GetGridHtml(this HtmlHelper html, IPublishedContent contentItem, string propertyAlias, string framework)
         {
-            if (string.IsNullOrWhiteSpace(propertyAlias)) throw new ArgumentNullOrEmptyException(nameof(propertyAlias));
+            if (propertyAlias == null) throw new ArgumentNullException(nameof(propertyAlias));
+            if (string.IsNullOrWhiteSpace(propertyAlias)) throw new ArgumentException("Value can't be empty or consist only of white-space characters.", nameof(propertyAlias));
 
             var view = "Grid/" + framework;
             var prop = contentItem.GetProperty(propertyAlias);
-            if (prop == null) throw new NullReferenceException("No property type found with alias " + propertyAlias);
+            if (prop == null) throw new NullReferenceException("No property type found with alias " + propertyAlias); // TODO NullReferenceException should not be thrown by user code
```

Comment:
We should probably throw an `InvalidOperationException` instead.

Reply:
Yep I agree